sdk/core/azure-core/src/http/request.cpp

```diff
@@ -10,18 +10,12 @@
 
 using namespace Azure::Core::Http;
 
-void Request::AddHeader(std::string const& name, std::string const& value)
+size_t Request::AddHeader(std::string const& name, std::string const& value)
```

Comment:
Same here/elsewhere. Instead of returning `size_t`, we should consider throwing an exception for invalid input.